### PR TITLE
[operator-trivy] add reportResourceLabels

### LINKED ISSUE #8972: [trivy] Add configuration option to Include custom labels in metrics
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

In some cases, there are too many metrics or teams to manage efficiently. Adding a configuration option in Trivy to include custom labels in metrics can help organize and analyze data more effectively.

### Proposed Solution

1. Use the existing function to add custom labels to metrics as described in the [trivy documentation](https://aquasecurity.github.io/trivy-operator/v0.6.0/integrations/metrics/#adding-custom-label-to-metrics)
2. Include an additional filter in the "Trivy Image Vulnerability Overview" dashboard to enhance usability and analytical capabilities. 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
This PR adds the [reportResourceLabels](https://aquasecurity.github.io/trivy-operator/v0.6.0/integrations/metrics/#adding-custom-label-to-metrics) option to operator-trivy configuration.

## Why do we need it, and what problem does it solve?
The reportResourceLabels option is a handy way of matching workloads being scanned to the respective Trivy reports.
Closes #8972 

## What is the expected result?
The option appears in the operator's configmap and new Trivy reports can be filtered by the set labels.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: operator-trivy
type: feature
summary: Add reportResourceLabels option.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by miklezzzz** [comment]
before
```
apiVersion: v1
data:
  configAuditReports.scanner: Trivy
  node.collector.imagePullSecret: deckhouse-registry
  node.collector.imageRef: dev-registry.deckhouse.io/sys/deckhouse-oss@sha256:e85813a2be2adf3fb9d61541481bcc15b8338e5b25afc047343379e59e90892e
  report.recordFailedChecksOnly: "true"
  scanJob.compressLogs: "false"
  skipResourceByLabels: upmeter-group,upmeter-probe
  trivy.serverURL: http://trivy-server.d8-operator-trivy.svc.cluster.local:4954
  vulnerabilityReports.scanner: Trivy

```
image

**Comment by miklezzzz** [comment]
after
```
apiVersion: v1
data:
  configAuditReports.scanner: Trivy
  node.collector.imagePullSecret: deckhouse-registry
  node.collector.imageRef: dev-registry.deckhouse.io/sys/deckhouse-oss@sha256:e85813a2be2adf3fb9d61541481bcc15b8338e5b25afc047343379e59e90892e
  report.recordFailedChecksOnly: "true"
  report.resourceLabels: app,env
  scanJob.compressLogs: "false"
  skipResourceByLabels: upmeter-group,upmeter-probe
  trivy.serverURL: http://trivy-server.d8-operator-trivy.svc.cluster.local:4954
  vulnerabilityReports.scanner: Trivy
```
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/gcp'...

File: ee/modules/500-operator-trivy/openapi/doc-ru-config-values.yaml [hunk 1]
```diff
@@ -28,3 +28,7 @@ properties:
       Структура, аналогичная `spec.nodeSelector` пода Kubernetes.
 
       Если ничего не указано или указано `false`, будет [использоваться автоматика](https://deckhouse.ru/documentation/v1/#выделение-узлов-под-определенный-вид-нагрузки).
+
+  reportResourceLabels:
+    description: |
+      Список дополнительных меток, значения которых будут соответствовать значениям меток сканируемых ресурсов, для маркировки отчетов (VulnerabilityReport) Trivy.
```
Comment (z9r5) [inline ee/modules/500-operator-trivy/openapi/doc-ru-config-values.yaml:34]:
```suggestion
      Список дополнительных меток маркировки отчетов (VulnerabilityReport) Trivy.
      
      Значения этих меток будут соответствовать значениям меток сканируемых ресурсов.
```

File: ee/modules/500-operator-trivy/openapi/config-values.yaml [hunk 2]
```diff
@@ -73,3 +73,12 @@ properties:
       The same as `spec.nodeSelector` for the Kubernetes pod.
 
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/documentation/v1/#advanced-scheduling).
+  reportResourceLabels:
+    type: array
+    description: |
+      A list of additional labels, which values are exported from the resources being scanned, to apply to Trivi's reports.
```
Comment (z9r5) [inline ee/modules/500-operator-trivy/openapi/config-values.yaml:79]:
```suggestion
      A list of additional labels for marking Trivi's reports (VulnerabilityReport).
      
      The values of these labels will correspond to the values of the scanned resources' labels.
```

**Review by yalosev — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Pull request merged**